src/tabs/ProgrammableREPL/index.tsx

```diff
@@ -0,0 +1,106 @@
+/**
+ * Tab for Source Academy Programmable REPL module
+ * @module programmable_repl
+ * @author Wang Zihan
+ */
+
+import React from 'react';
+import { DebuggerContext } from '../../typings/type_helpers';
+import { Button, ButtonGroup, Divider, NumericInput } from '@blueprintjs/core';
+import { IconNames } from '@blueprintjs/icons';
+import { ProgrammableRepl } from '../../bundles/programmable_repl/programmable_repl';
+import AceEditor from 'react-ace';
+import 'ace-builds/src-noconflict/mode-javascript';
+import 'ace-builds/src-noconflict/theme-twilight';
+import 'ace-builds/src-noconflict/ext-language_tools';
+
+
+type GUI_Data = {
+  programmableReplInstance: ProgrammableRepl;
+};
+
+class ProgrammableReplGUI extends React.Component<GUI_Data> {

```

Comment:
The convention is to name it `***Props`, but it's alright, just need to change the casing.